core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -11,149 +11,173 @@ import "../Finder.sol";
-     */
+    // Maps sponsor addresses to their positions. Each sponsor can have only one position.
     mapping(address => PositionData) public positions;
 
     // Keep track of the total collateral and tokens across all positions to enable calculating the global
     // collateralization ratio without iterating over all positions.
     FixedPoint.Unsigned public totalPositionCollateral;
     FixedPoint.Unsigned public totalTokensOutstanding;
 
-    ExpandedIERC20 public tokenCurrency;
-    IERC20 public collateralCurrency;
-
-    // Finder knows the deployed addresses of all other smart contracts in UMA ecosystem.
-    Finder public finder;
+    ExpandedIERC20 public tokenCurrency; // Synthetic token created by this contract
+    IERC20 public collateralCurrency; // Collateral held by the contract to back synthetics.
+    Finder public finder; // Finder knows the deployed addresses of all other smart contracts in UMA ecosystem.
 
     // Unique identifier for DVM price feed ticker.
-    bytes32 priceIdentifer;
-
-    /**
-     * @dev Time that this contract expires.
-     */
+    bytes32 public priceIdentifer;
+    // Time that this contract expires.
     uint public expirationTimestamp;
-    /**
-     * @dev Time that has to elapse for a withdrawal request to be considered passed, if no liquidations occur.
-     */
+    // Time that has to elapse for a withdrawal request to be considered passed, if no liquidations occur.
     uint public withdrawalLiveness;
 
+    event Transfer(address indexed oldSponsor, address indexed newSponsor);
+    event Deposit(address indexed sponsor, uint indexed collateralAmount);
+    event Withdrawal(address indexed sponsor, uint indexed collateralAmount);
+    event RequestWithdrawal(address indexed sponsor, uint indexed collateralAmount);
+    event RequestWithdrawalExecuted(address indexed sponsor, uint indexed collateralAmount);
+    event RequestWithdrawalCanceled(address indexed sponsor, uint indexed collateralAmount);
+    event PositionCreated(address indexed sponsor, uint indexed collateralAmount, uint indexed tokenAmount);
+    event Redeem(address indexed sponsor, uint indexed collateralAmount, uint indexed tokenAmount);
+    event ContractExpired(address indexed caller);
+    event SettleExpiredPosition(
+        address indexed caller,
+        uint indexed collateralAmountReturned,
+        uint indexed tokensBurned
+    );
+
+    modifier onlyPreExpiration() {
+        require(getCurrentTime() < expirationTimestamp, "Cannot operate on a position past its expiry time");
+        _;
+    }
+
+    modifier onlyPostExpiration() {
+        require(getCurrentTime() >= expirationTimestamp, "Cannot operate on a position before its expiry time");
+        _;
+    }
+
+    modifier onlyCollateralizedPosition(address sponsor) {
+        require(positions[sponsor].collateral.rawValue > 0, "Position has no collateral and so is invalid");
```

Comment:
More for my own understanding, do we pay multiple storage read costs if we read the same fields in this modifier and then again in each method?

Reply:
I actually have no idea. @mrice32 ? I will look into this as well

Reply:
Yes, I think we do. I can't find any good references for this, but as far as I'm aware each SLOAD opcode gets charged 800 gas. The SSTORE opcode is the only one that gets special net-metering treatment. The only thing that might save us is if the solidity optimizer is smart and caches the value in memory (I highly doubt it's that smart, though).

Reply:
Interesting, thanks for researching. We don't care about minimizing gas costs at this stage, so nothing to change here. I think the modifier is a win for readability.